ServerlessLibraryAPI/CacheService.cs

```diff
@@ -88,6 +88,10 @@ private async Task<IList<LibraryItem>> ConstructCache()
                     {
                         foreach (LibraryItem libraryItem in libraryItems)
                         {
+                            int itemsCountInStore = this.libraryStore.Count();
+
+                            // Assign an Id which is a string form of integer to each item. Otherwise cosmosdb will assign a guid as id which is not user friendly.  
+                            libraryItem.Id = (itemsCountInStore + 1) + string.Empty;
```

Comment:
This may not be ideal with mulitple parallel calls. We dont have traffic at this time that would cause this, and I presume a retry woudl work. But investigate cosmosdb triggers to increment this value perhaps ?https://stackoverflow.com/a/27785611/5889849

Reply:
I understand that concurrent add items will fail with current implementation and we need to handle error in case of concurrency errors. Probably user needs to retry if add item fails or we put api may retry couple of times.

The link you suggested seems to have issues too.

Reply:
With trigger approach or store procedure approach , still we have concurrency concerns. How do we ensure that two different inserts same time will not read same value of the counter?